etl/steps/data/explorers/migration/2024-08-05/refugees_asylum_migration.py

```diff
@@ -0,0 +1,239 @@
+
+    #
+    # Process data.
+    #
+    # Prepare graphers table of explorer.
+    graphers_dicts = []
+
+    for tb, ds in tbs_and_ds:
+        graphers_dicts = create_graphers_rows(graphers_dicts, tb, ds)
+
+    df_graphers = pd.DataFrame(graphers_dicts)
+
+    # Add a map tab to all indicators.
+    df_graphers["hasMapTab"] = "true"
+    # show map tab by default
+    df_graphers["tab"] = "map"
+    # set yAxis to start at 0
+    df_graphers["yAxisMin"] = 0
+    # hide automatic time/ entity in chart title
+    df_graphers["hideAnnotationFieldsInTitle"] = "true"
+    # set current year als maximum year
+    df_graphers["timelineMaxTime"] = 2024
+
+    # Sanity check.
+    error = "Duplicated rows in explorer."
+    assert df_graphers[
+        df_graphers.duplicated(subset=["Metric Dropdown", "Period Radio", "Sub-Metric Radio", "Age Radio"], keep=False)
+    ].empty, error
+
+    # Sort rows conveniently
+    df_graphers["sort_order_metrics"] = df_graphers["Metric Dropdown"].apply(sort_metrics)
+    df_graphers["sort_order_age"] = df_graphers["Age Radio"].apply(sort_age)
+    df_graphers["sort_order_period"] = df_graphers["Period Radio"].apply(sort_period)
+    df_graphers["sort_order_sub_metric"] = df_graphers["Sub-Metric Radio"].apply(sort_sub_metric)
+
+    df_graphers = df_graphers.sort_values(
+        by=["sort_order_metrics", "sort_order_age", "sort_order_period", "sort_order_sub_metric"],
+        ascending=True,
+    ).reset_index(drop=True)
+
+    df_graphers = df_graphers.drop(
+        columns=["sort_order_metrics", "sort_order_age", "sort_order_period", "sort_order_sub_metric"]
+    )
+
+    # Prepare explorer metadata.
+    config = {
+        "explorerTitle": "Migration, Refugees, and Asylum Seekers",
+        "explorerSubtitle": "Explore the migration of people across the world.",
+        "selection": [
+            "Canada",
+            "France",
+            "Germany",
+            "United Kingdom",
+            "United States",
+            "India",
+            "China",
+            "Syria",
+            "Yemen",
+        ],
+    }
+
+    # create columns for the explorer
+    col_dicts = []
+
+    for tb, ds in tbs_and_ds:
+        col_dicts = create_column_rows(col_dicts, tb, ds)
+
+    df_columns = pd.DataFrame(col_dicts)
+
+    df_columns["colorScaleNumericMinValue"] = 0
+    df_columns["colorScaleEqualSizeBins"] = "true"
+
+    # Save outputs.
+    ds_explorer = create_explorer(dest_dir=dest_dir, config=config, df_graphers=df_graphers, df_columns=df_columns)
+    ds_explorer.save()
+
+
+def create_graphers_rows(graphers_dicts, tb, ds):
+    # read out of list
+    for column in tb.drop(columns=["country", "year"]).columns:
+        if tb[column].notnull().any():
+            if column not in CONFIG_DICT.keys():
+                continue
+            graphers_row_dict = {}
+
+            meta = tb[column].metadata
+            origin = meta.origins[0]
```

Comment:
Not really, I was just under the impression that the explorer file would somehow overwrite the sources and FASTT if I don't define them explicitly. Good to know that's not the case! 

I kept the population origin in now - do you see any problem with that in general? I think it is even more thorough this way.

Reply:
If they don't bother you, then keep all origins :) . That simplifies the code and ensures all origins are propagated automatically, so you won't need to make manual changes in the future, in case something changes in the data (e.g. the order of origins is reversed for some reason).

Reply:
That sounds good - I'll keep it like this then. Thanks Pablo! :D